sunraster JOSS paper. (#231)
sunpy/sunraster: joss_paper/paper.md

 - name: Bay Area Environmental Research Institute, USA
   index: 4
 - name: Aperio Software Ltd, UK
   index: 5
 - name: Claude Bernard Lyon 1 University, France
   index: 6
 - name: Institute of Theoretical Astrophysics, Norway
   index: 7
 - name: NASA Goddard Space Flight Center, USA
   index: 8
 - name: Workato Gmbh, Germany
   index: 9
 - name: Université Paris-Saclay, CNRS, Institut d'Astrophysique Spatiale, France
   index: 10
date: 24 February 2023
bibliography: paper.bib

---

# Summary

sunraster is a free, open-source, community-developed Python package for inspecting,
manipulating, and visualizing solar rastering slit-spectrograph observations.
The sunraster data classes link observations with various forms of supporting data
including: measurement uncertainties; units; a data mask to mark pixels with unreliable
or un-physical data values; WCS (World Coordinate System) transformations that describe
the position, wavelengths and times represented by the pixels; and general metadata.
These classes provide support for analysis tasks, such as extracting regions of
interest and applying/removing exposure time corrections, as well as a quick-look
visualization suite.

> sunraster depends heavily on the ndcube package [@ndcube-apj, @ndcube-joss] and its

Author: ```suggestion
sunraster depends heavily on the ndcube package [@ndcube-apj; @ndcube-joss] and its
```

SpectrogramCube class complies with the NDCube 2 API [sep12; @ndcube-apj].

# Statement of Need

Observing the location, morphology, dynamics and energetics of physical phenomena

> are often key to understanding the underlying physics that drives them.

Author: Indeed you are right!

```suggestion
is often key to understanding the underlying physics that drives them.
```

However, making simultaneous spatial, temporal and spectral measurements with a single
instrument is often challenging.

> One solution is to disperse the incoming light through a narrow slit onto a pixelated

Author: ```suggestion
One option is to disperse the incoming light through a narrow slit onto a pixelated
```

Orbiter [SPICE; @spice].

However the rastering slit-spectrograph design leads to a fundamental complication,
namely that one of the spatial dimensions is coupled with time.
This can create ambiguity if the dynamic timescales of the observed phenomenon is
less than or comparable to the rastering cadence.
This requires scientists to think of their data as both 3-D (time/space-spectral-space)
and 4-D (time-spectral-space-space/time) alternately, and sometimes simultaneously,
depending on their science goals.

By building on features inherited from the ndcube package, sunraster provides an API
for seamlessly switching between the 3-D and 4-D paradigms without changing or
hard-copying the underlying data.
This saves users carrying around multiple copies of the same data which might become
inconsistent during a data analysis workflow.
A powerful slicing API enables users to extract regions of interest with single commands
that not only alter the data array, but also the associated uncertainties, data mask
and the coordinate transformations.
This saves users from performing multiple tedious and error-prone tasks on the various
compoennts of their dataset individually.

> This makes the development of analysis workflows quicker, more reliable, and easier

Reviewer: "It also makes"
```suggestion
This makes the development of analysis workflows quicker, more reliable, and easier
```

to understand, thus boosting users' productivity.

# Community Applications

> Most of sunrasters tools are instrument-agnostic.

Author: ```suggestion
Most of sunraster's tools are instrument-agnostic.
```

However it does provide specific tools for reading Solar Orbiter/SPICE FITS files
and hence helping SPICE users leverage the wider scientific Python ecosystem
for their analysis.

> In addition, the Python user tools package for the IRIS mission, irispy-lmsal
> [irispy-docs; @irispy-code], depends on sunraster.

Author: How about this @ebuchlin 
```suggestion
In addition, the data and metadata classes in the IRIS mission's Python user tools package [irispy-lmsal;
@irispy-docs; @irispy-code] inherit from and build upon those provided by sunraster.
```

# Acknowledgements

We acknowledge financial support for sunraster from Solar Orbiter/SPICE

> (grant 80NSSC19K1000).

Author: ```suggestion
(grant 80NSSC19K1000) as well as NASA's SDO/AIA and IRIS missions
(grant NNG04EA00C and NNG09FA40C).
```

We also acknowledge the SunPy and Python in Heliophysics communities for their support.

> 

Reviewer 2: ```suggestion
We acknowledge financial support for sunraster from NASA's SDO/AIA and IRIS missions
(grant NNG04EA00C and NNG09FA40C).
```

Author: Moving this to be part of first sentence.

Orbiter [SPICE; @spice].

However the rastering slit-spectrograph design leads to a fundamental complication,
namely that one of the spatial dimensions is coupled with time.
This can create ambiguity if the dynamic timescales of the observed phenomenon is
less than or comparable to the rastering cadence.
This requires scientists to think of their data as both 3-D (time/space-spectral-space)
and 4-D (time-spectral-space-space/time) alternately, and sometimes simultaneously,
depending on their science goals.

By building on features inherited from the ndcube package, sunraster provides an API
for seamlessly switching between the 3-D and 4-D paradigms without changing or
hard-copying the underlying data.
This saves users carrying around multiple copies of the same data which might become
inconsistent during a data analysis workflow.
A powerful slicing API enables users to extract regions of interest with single commands
that not only alter the data array, but also the associated uncertainties, data mask
and the coordinate transformations.
This saves users from performing multiple tedious and error-prone tasks on the various

> compoennts of their dataset individually.

Reviewer 2: ```suggestion
components of their dataset individually.
```
